File: lib/features/pharmacy/domain/models/pharmacy.dart
class Pharmacy {
  final String id;
  final String name;
  final String logoUrl;
  final bool isActive;
  final String address;

  Pharmacy({
    required this.id,
    required this.name,
    required this.logoUrl,
    this.isActive = true,
    this.address = '',
  });

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'logoUrl': logoUrl,
    'isActive': isActive,
    'address': address,
  };
}

File: backend/src/models/Pharmacy.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IPharmacy extends Document {
  name: string;
  address: string;
  logoUrl?: string;
  phone?: string;
  isActive: boolean;
  serviceEnabled: boolean;
}

const pharmacySchema = new Schema<IPharmacy>(
  {
    name: { type: String, required: true, trim: true },
    address: { type: String, required: true },
    logoUrl: { type: String },
    phone: { type: String },
    isActive: { type: Boolean, default: true },
    serviceEnabled: { type: Boolean, default: true },
  },
  { timestamps: true, collection: 'pharmacies' },
);

export const Pharmacy = mongoose.model<IPharmacy>('Pharmacy', pharmacySchema);
